### ISSUE TITLE
Update `@yamada-ui/select` accessibility

### ISSUE THREAD

**Issue by nakayan5** [yamada-ui/yamada-ui#1959]
### Description

This issue aims to address accessibility deficiencies in our select component as per W3C guidelines.
The component currently does not meet the accessibility standards outlined in the W3C documentation.

The goal is to ensure the component is accessible to all users, including those using assistive technologies.
Ensure that all unchecked items in this checklist should be addressed.

Keyboard interaction: none

WAI-ARIA Roles, States, and Properties
- [x] An element that contains or owns all the listbox options has role [listbox](https://w3c.github.io/aria/#listbox).
- [x] Each option in the listbox has role [option](https://w3c.github.io/aria/#option) and is contained in or owned by either:
  - [x] The element with role listbox.
  - [x] An element with role [group](https://w3c.github.io/aria/#group) that is contained in or owned by the element with role listbox.
- [x] Options contained in a group are referred to as grouped options and form what is called an option group. If a listbox contains grouped options, then:
  - [x] All option groups contain at least one option.
  - [x] Each option group has an accessible name provided via [aria-label](https://w3c.github.io/aria/#aria-label) or [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby).
- [ ] If the element with role listbox is not part of another widget, such as a combobox, then it has either a visible label referenced by [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) or a value specified for [aria-label](https://w3c.github.io/aria/#aria-label).
- [x] If the listbox supports selection of more than one option, the element with role listbox has [aria-multiselectable](https://w3c.github.io/aria/#aria-multiselectable) set to true. Otherwise, aria-multiselectable is either set to false or the default value of false is implied.
- [x] The selection state of each selectable option is indicated with either [aria-selected](https://w3c.github.io/aria/#aria-selected) or [aria-checked](https://w3c.github.io/aria/#aria-checked):
  - [x] If the selection state is indicated with aria-selected, then aria-checked is not specified for any options. Alternatively, if the selection state is indicated with aria-checked, then aria-selected is not specified for any options. See notes below regarding considerations for which property to use and for details of the unusual conditions that might allow for both properties in the same listbox.
  - [x] If any options are selected, each selected option has either [aria-selected](https://w3c.github.io/aria/#aria-selected) or [aria-checked](https://w3c.github.io/aria/#aria-checked) set to true. No more than one option is selected at a time if the element with role listbox does not have [aria-multiselectable](https://w3c.github.io/aria/#aria-multiselectable) set to true.
  - [x] All options that are selectable but not selected have either [aria-selected](https://w3c.github.io/aria/#aria-selected) or [aria-checked](https://w3c.github.io/aria/#aria-checked) set to false.
  - [x] Note that except in listboxes where selection follows focus, the selected state is distinct from focus. For more details, see [this description of differences between focus and selection](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#kbd_focus_vs_selection) and [Deciding When to Make Selection Automatically Follow Focus](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#kbd_selection_follows_focus).
- [ ] If the complete set of available options is not present in the DOM due to dynamic loading as the user scrolls, their [aria-setsize](https://w3c.github.io/aria/#aria-setsize) and [aria-posinset](https://w3c.github.io/aria/#aria-posinset) attributes are set appropriately.
- [ ] If options are arranged horizontally, the element with role listbox has [aria-orientation](https://w3c.github.io/aria/#aria-orientation) set to horizontal. The default value of aria-orientation for listbox is vertical.

### Problem Statement/Justification

improve accessibility

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

### Additional Information

related https://github.com/yamada-ui/yamada-ui/issues/1500

**Comment by uraaaa24** [comment]
> An element that contains or owns all the listbox options has role [listbox](https://w3c.github.io/aria/#listbox).

Already supported.
https://github.com/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L776

**Comment by uraaaa24** [comment]
> Each option in the listbox has role [option](https://w3c.github.io/aria/#option) and is contained in or owned by either:
 The element with role listbox.
 An element with role [group](https://w3c.github.io/aria/#group) that is contained in or owned by the element with role listbox.

Already supported.
https://github.com/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L1021

**Comment by uraaaa24** [comment]
> Options contained in a group are referred to as grouped options and form what is called an option group. If a listbox contains grouped options, then:
 All option groups contain at least one option.
 Each option group has an accessible name provided via [aria-label](https://w3c.github.io/aria/#aria-label) or [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby).

Already supported.
https://github.com/yamada-ui/yamada-ui/blob/main/packages/components/select/src/select.tsx#L148

**Comment by uraaaa24** [comment]
>  The selection state of each selectable option is indicated with either [aria-selected](https://w3c.github.io/aria/#aria-selected) or [aria-checked](https://w3c.github.io/aria/#aria-checked):
If the selection state is indicated with aria-selected, then aria-checked is not specified for any options. Alternatively, if the selection state is indicated with aria-checked, then aria-selected is not specified for any options. See notes below regarding considerations for which property to use and for details of the unusual conditions that might allow for both properties in the same listbox.
All options that are selectable but not selected have either [aria-selected](https://w3c.github.io/aria/#aria-selected) or [aria-checked](https://w3c.github.io/aria/#aria-checked) set to false.

Already supported.
https://github.com/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L1030

> Note that except in listboxes where selection follows focus, the selected state is distinct from focus. For more details, see [this description of differences between focus and selection](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#kbd_focus_vs_selection) and [Deciding When to Make Selection Automatically Follow Focus](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#kbd_selection_follows_focus).

Already supported.
https://github.com/yamada-ui/yamada-ui/blob/main/packages/components/select/src/use-select.ts#L1028

> If any options are selected, each selected option has either [aria-selected](https://w3c.github.io/aria/#aria-selected) or [aria-checked](https://w3c.github.io/aria/#aria-checked) set to true. No more than one option is selected at a time if the element with role listbox does not have [aria-multiselectable](https://w3c.github.io/aria/#aria-multiselectable) set to true.

This pull request will be addressed in this.
https://github.com/yamada-ui/yamada-ui/pull/2800

**Comment by uraaaa24** [comment]
> If the element with role listbox is not part of another widget, such as a combobox, then it has either a visible label referenced by [aria-labelledby](https://w3c.github.io/aria/#aria-labelledby) or a value specified for [aria-label](https://w3c.github.io/aria/#aria-label).
If the complete set of available options is not present in the DOM due to dynamic loading as the user scrolls, their [aria-setsize](https://w3c.github.io/aria/#aria-setsize) and [aria-posinset](https://w3c.github.io/aria/#aria-posinset) attributes are set appropriately.
If options are arranged horizontally, the element with role listbox has [aria-orientation](https://w3c.github.io/aria/#aria-orientation) set to horizontal. The default value of aria-orientation for listbox is vertical.

In this select component, out of scope.

**Comment by uraaaa24** [comment]
> If the listbox supports selection of more than one option, the element with role listbox has [aria-multiselectable](https://w3c.github.io/aria/#aria-multiselectable) set to true. Otherwise, aria-multiselectable is either set to false or the default value of false is implied.

This pull request completes the response.
https://github.com/yamada-ui/yamada-ui/pull/2800
